src/collect/collector.rs

```diff
@@ -137,22 +150,22 @@ impl Collectors {
         }
 
         // Setup user defined probes.
-        for probe in collect.args()?.probes.iter() {
-            self.probes.add_probe(self.parse_probe(probe)?)?;
-        }
-
-        self.setup_filters(collect)?;
+        let mut probes = Vec::new();
+        collect.args()?.probes.iter().try_for_each(|p| {
+            probes.push(self.parse_probe(p)?);
+            Ok::<(), anyhow::Error>(())
+        })?;
+        probes
+            .drain(..)
+            .try_for_each(|p| self.probes.add_probe(p))?;
```

Comment:
Adding cli as member of `Collectors` forces us to do this two-step probe adding.

Considering my previous experiment, would you consider this other approach?
https://github.com/amorenoz/retis/commit/5cd386e8c0a8fdd5da2490f0e63535ff353b2e95

TL;DR: Instead of passing cli in `Collectors::new()`, pass it in a new `Collectors::run()` that does all the processing. That way, we still own the cli but it's not a member of `Collectors`.

Note: I'm not suggesting adding it to this PR, I'm just considering this direction. If you think it's ok I'll fix the patch and send it.

Reply:
> I'm wondering if we could avoid using the cli obj as an argument there though, it seems quite complex to take a subcommand out of cli and then use cli as an argument for all the runner helpers. There might be a way to make all this more self-contained, but honestly haven't looked at this so this might be OK this way. (The above makes sense regardless of this).

That's precisely why I ended up having to split the Runner from the SubCommand itself.